roles/ovirt/tasks/create_vms.yml

```diff
@@ -0,0 +1,84 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Generate list of networks
+  when: item.nets | length > 0
+  ansible.builtin.set_fact:
+    _nics_list: "{{ _nics_list | default([]) + [{'name': 'nic' ~ (index + 1) , 'profile_name': inner_item}] }}"
+  loop: "{{ item.nets }}"
+  loop_control:
+    index_var: index
+    loop_var: inner_item
+
+- name: Generate VM name with prefix
+  when: cifmw_ovirt_vm_name_prefix is defined
+  ansible.builtin.set_fact:
+    _vm_name: "{{ cifmw_ovirt_vm_name_prefix }}-{{ cifmw_ovirt_env_prefix }}-{{ item.name }}"
+
+- name: Generate VM name
+  when: cifmw_ovirt_vm_name_prefix is not defined
+  ansible.builtin.set_fact:
+    _vm_name: "{{ cifmw_ovirt_env_prefix }}-{{ item.name }}"
+
+- name: Gather vnic profiles info
+  loop: "{{ _nics_list }}"
+  loop_control:
+    loop_var: inner_item
+  register: _network_info
+  ovirt.ovirt.ovirt_vnic_profile_info:
+    auth: "{{ ovirt_auth }}"
+    name: "{{ inner_item.profile_name }}"
+
+- name: Fail if any of the vnic profiles does not exist
+  when: item.ovirt_vnic_profiles | length == 0
+  loop: "{{ _network_info.results }}"
+  ansible.builtin.fail:
+    msg: "Network {{ item.item.profile_name }} does not exist"
+
+- name: Create VMs from the imported template
+  ovirt.ovirt.ovirt_vm:
+    auth: "{{ ovirt_auth }}"
+    name: "{{ _vm_name }}{{ '%02d' | format(ansible_loop.index) }}"
+    type: server
+    cluster: "{{ cifmw_ovirt_cluster_name }}"
+    template: "{{ item.template }}"
+    cpu_cores: "{{ item.cpu | default ('1') }}"
+    memory: "{{ item.memory | default ('4') }}GiB"
+    state: "{{ cifmw_ovirt_vm_state }}"
+    comment: "{{ cifmw_ovirt_vm_comment }}"
+    clone: yes
+    nics: "{{ _nics_list }}"
+    wait: true
+    timeout: 300
+    cloud_init:
+      authorized_ssh_keys: >-
+        {{
+          lookup('file', ansible_user_dir + '/.ssh/cifmw_reproducer_key.pub') |
```

Comment:
IMHO, we should decouple this from `reproducer`. It is good to look for the public key... if not found generate one.

Secondly, instead of `lookup`, opt to use `slurp` as there is a higher probability of it being executed remotely.

Reply:
Done, thanks.